src/Collection/CollectionTrait.php

```diff
@@ -135,10 +135,16 @@ public function map(callable $c)
      * {@inheritDoc}
      *
      */
-    public function reduce(callable $c, $zero)
+    public function reduce(callable $c, $zero = null)
     {
+        $collection = $this;
+        if ($zero === null) {
+            $zero = $this->first();
+            $collection = $this->take(iterator_count($this) - 1, 1);
+        }
```

Comment:
This has the disadvantage that non-rewindable collections will throw an exception. I would change this by using a simple `if ($isFirst)`  inside the foreach.